be/src/vec/exec/vpartition_sort_node.cpp

```diff
@@ -38,23 +39,59 @@
 #include "vec/exprs/vexpr_context.h"
 
 namespace doris::vectorized {
-Status PartitionBlocks::do_partition_topn_sort() {
+Status PartitionBlocks::append_block_by_selector(const vectorized::Block* input_block, bool eos) {
+    if (blocks.empty() || reach_limit()) {
+        _init_rows = _partition_sort_info->_runtime_state->batch_size();
+        blocks.push_back(Block::create_unique(
+                VectorizedUtils::create_empty_block(_partition_sort_info->_row_desc)));
+    }
+    auto columns = input_block->get_columns();
+    auto mutable_columns = blocks.back()->mutate_columns();
+    DCHECK(columns.size() == mutable_columns.size());
+    for (int i = 0; i < mutable_columns.size(); ++i) {
+        columns[i]->append_data_by_selector(mutable_columns[i], selector);
+    }
+    blocks.back()->set_columns(std::move(mutable_columns));
+    auto selector_rows = selector.size();
+    _init_rows = _init_rows - selector_rows;
+    _total_rows = _total_rows + selector_rows;
+    _current_input_rows = _current_input_rows + selector_rows;
+    selector.clear();
+    // maybe better could change by user PARTITION_SORT_ROWS_THRESHOLD
+    if (!eos && _partition_sort_info->_partition_inner_limit != -1 &&
+        _current_input_rows >= PARTITION_SORT_ROWS_THRESHOLD &&
+        _partition_sort_info->_topn_phase != TPartTopNPhase::TWO_PHASE_GLOBAL) {
+        create_or_reset_sorter_state();
+        RETURN_IF_ERROR(do_partition_topn_sort());
+        _current_input_rows = 0; // reset record
```

Comment:
0? may rows still in `PartitionBlocks`

Reply:
this is used to get filter rows then do topn: filter = _current_input_rows - current_output_rows.
so need reset to 0 after do sort.